exetera/core/dataset.py

```diff
@@ -69,7 +20,9 @@ def __init__(self, session, dataset_path, mode, name):
         self._session = session
         self.file = h5py.File(dataset_path, mode)
         self.dataframes = dict()
```

Comment:
I think it would be a good idea to have self.dataframes become self._dataframes for internal usage, and for us to add a property dataframes that copies the dictionary so that manipulating the dictionary doesn't change the true internal state of the dataset. I should have done the same thing with Sessions.datasets.